flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/dsv2/windowing/CountProductSalesWindowing.java

```diff
@@ -0,0 +1,235 @@
+                                        new CountSalesQuantity()));
+
+        if (fileOutput) {
+            productSalesQuantityStream
+                    .toSink(
+                            new WrappedSink<>(
+                                    FileSink.<Tuple3<Integer, Long, Long>>forRowFormat(
+                                                    new Path(params.get("output")),
+                                                    new SimpleStringEncoder<>())
+                                            .withRollingPolicy(
+                                                    DefaultRollingPolicy.builder()
+                                                            .withMaxPartSize(
+                                                                    MemorySize.ofMebiBytes(1))
+                                                            .withRolloverInterval(
+                                                                    Duration.ofSeconds(10))
+                                                            .build())
+                                            .build()))
+                    .withName("file-sink");
+        } else {
+            // Print the results to the STDOUT.
+            productSalesQuantityStream
+                    .toSink(new WrappedSink<>(new PrintSink<>()))
+                    .withName("print-sink");
+        }
+
+        env.execute("CountProductSalesWindowing example");
+    }
+
+    /**
+     * Count sales quantity per product.
+     *
+     * <p>We will obtain all orders and calculate the total sales quantity for each product when
+     * window trigger.
+     */
+    public static class CountSalesQuantity
+            implements OneInputWindowStreamProcessFunction<
+                    Tuple2<Integer, Long>, Tuple3<Integer, Long, Long>> {
+
+        @Override
+        public void onTrigger(
+                Collector<Tuple3<Integer, Long, Long>> output,
+                PartitionedContext<Tuple3<Integer, Long, Long>> ctx,
+                OneInputWindowContext<Tuple2<Integer, Long>> windowContext)
+                throws Exception {
+            // get current productId
+            int productId = ctx.getStateManager().getCurrentKey();
+            // calculate total sales quantity
+            long totalSalesQuantity = 0;
+            for (Tuple2<Integer, Long> ignored : windowContext.getAllRecords()) {
+                totalSalesQuantity += 1;
+            }
+            // emit result
+            output.collect(Tuple3.of(productId, windowContext.getStartTime(), totalSalesQuantity));
+        }
+    }
+
+    /**
+     * Count sales quantity per product.
+     *
+     * <p>Firstly, we declare a state to store the sales quantity for each product.
+     *
+     * <p>When receive a record, we update the sales quantity in state.
+     *
+     * <p>When window trigger, we emit the result.
+     */
+    public static class CountSalesQuantityWithPreAggregation
```

Comment:
I think we could remove this unused function. User can learn how to use pre-agg window from documentation.